tools/check_volume_levels.m

```diff
@@ -187,15 +187,28 @@ function plot_levels(meas, tc, lm)
 
 function pass = check_levels(meas, tc, lm, verbose)
 	pass = 1;
+	dg_tol = 0.1;
 	gains =  meas.levels - lm.sine_dbfs;
 	sv = size(tc.volumes);
 	for j = 1:sv(2)
 		for i = 1:sv(1)
+			% Initial location to test
 			ts = tc.vctimes(i)+tc.meas(1);
 			te = tc.vctimes(i)+tc.meas(2);
 			idx0 = find(meas.t < te);
 			idx = find(meas.t(idx0) > ts);
-			avg_gain = mean(gains(idx, j));
+
+			% Delay if settled gain is later in the window,
+			% this adds more robustness to test for controls
+			% apply delay.
+			dg = diff(gains(idx,j));
+			if max(abs(dg)) > dg_tol
+				dg_rev = dg(end:-1:1);
+				idx_add = length(dg) - find(abs(dg_rev) > 0.1, 1, 'first') + 1;
+			else
+				idx_add = 0;
+			end
+			avg_gain = mean(gains(idx + idx_add, j));
```

Comment:
The idx_add may push the end of indices outside length of gains(), need to limit gain averaging to available samples amount.

Reply:
Fixed now